doc/article/en-US/templating-html-behaviors-introduction.md

```diff
@@ -0,0 +1,307 @@
+
+When you tell Aurelia to inject an instance of the `Element` class in to your class, Aurelia will inject the DOM Element your HTML Behavior is associated with. So no need to futz around with DOM queries! Let's take our `red-square` attribute and have it actually set it's element to be a red square. Just assign the passed in object to a property on your class, and you're good to go.
+
+<code-listing heading="red-square${context.language.fileExtension}">
+  <source-code lang="ES 2015">
+    export class RedSquareCustomAttribute {
+      static inject = [Element];
+
+      constructor(element){
+        this.element = element;
+
+        this.element.style.width = this.element.style.height = '100px';
+        this.element.backgroundColor = 'red';
+      }
+    }
+  </source-code>
+  <source-code lang="ES 2016">
+    import {inject} from 'aurelia-framework';
+
+    @inject(Element)
+    export class RedSquareCustomAttribute {
+
+      constructor(element){
+        this.element = element;
+
+        this.element.style.width = this.element.style.height = '100px';
+        this.element.backgroundColor = 'red';
+      }
+    }
+  </source-code>
+  <source-code lang="TypeScript">
+    import {autoinject} from 'aurelia-framework';
+
+    @autoinject
+    export class RedSquareCustomAttribute {
+      constructor(private element: Element){
+        this.element.style.width = this.element.style.height = '100px';
+        this.element.backgroundColor = 'red';
+      }
+    }
+  </source-code>
+</code-listing>
+
+Remember, you don't need jQuery to get the Element your HTML Behavior is attached to. Aurelia can provide it to you!
+
+## [Making an HTML Behavior Available in a View](aurelia-doc://section/4/version/1.0.0)
+
+So far, we've merely talked about creating HTML Behaviors, but creating a behavior isn't very useful if you can't use it. So how do you use one? Well, first we have to make it available in a view. HTML Behaviors (and other view resources) are not available globally by default. This means that you must let Aurelia know which resources will be available in each view. This is done using the `require` element. The `require` element has two attributes `from` and `as`. You *must* supply a value for the `from` attribute, while `as` is optional. You cannot use data-binding with the `from` or `as` attributes on `require`.  When you "require in" a view resource with a `require` element, Aurelia will make it available for use in your view.
+
+<code-listing heading="Require an External Resource">
+  <source-code lang="HTML">
+    <template>
+      <require from="my-custom-element"></require>
```

Comment:
@AshleyGrant , @EisenbergEffect , (I guess it doesn't affect the docs app, but) what do You think about using blank line as the first line inside the template, so that the content would be rendered properly also in Markdown preview? For example this code listing has lost the most important line when viewing it from Github: https://github.com/aurelia/templating/blob/master/doc/article/en-US/templating-html-behaviors-introduction.md#making-an-html-behavior-available-in-a-view